vector/src/main/java/im/vector/activity/LoginActivity.java

```diff
@@ -70,27 +63,29 @@
 import org.matrix.androidsdk.util.Log;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.Random;
 
+import fr.gouv.tchap.sdk.rest.client.TchapRestClient;
+import fr.gouv.tchap.sdk.rest.model.Platform;
 import im.vector.LoginHandler;
 import im.vector.Matrix;
-import im.vector.PhoneNumberHandler;
 import im.vector.R;
 import im.vector.RegistrationManager;
 import im.vector.UnrecognizedCertHandler;
 import im.vector.receiver.VectorRegistrationReceiver;
 import im.vector.receiver.VectorUniversalLinkReceiver;
 import im.vector.services.EventStreamService;
 import im.vector.util.DinsicUtils;
-import im.vector.util.PhoneNumberUtils;
 
 /**
  * Displays the login screen.
  */
 public class LoginActivity extends MXCActionBarActivity implements RegistrationManager.RegistrationListener, RegistrationManager.UsernameValidityListener {
+    public static final List<String> mSupportedHosts = Arrays.asList("matrix.i.tchap.rie.gouv.fr", "matrix.a.tchap.rie.gouv.fr", "matrix.e.tchap.rie.gouv.fr");
```

Comment:
for info, there is a config file where `matrix_org_server_url` is stored for example.